sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/AdoDtoHelper.java

```diff
@@ -61,62 +67,94 @@
 
 	protected abstract void fillInnerFromAdo(DTO dto, ADO ado);
 
-	protected void preparePulledResult(List<DTO> result) {
+	protected void preparePulledResult(List<DTO> result) throws NoConnectionException, ServerCommunicationException, ServerConnectionException, DaoException {
 	}
 
+	protected abstract long getApproximateJsonSizeInBytes();
+
 	/**
 	 * @return another pull needed?
+	 * @param context
 	 */
-	public boolean pullAndPushEntities() throws DaoException, ServerConnectionException, ServerCommunicationException, NoConnectionException {
+	public boolean pullAndPushEntities(Context context) throws DaoException, ServerConnectionException, ServerCommunicationException, NoConnectionException {
 
-		pullEntities(false);
+		pullEntities(false, context);
 
 		return pushEntities(false);
 	}
 
-	public void pullEntities(final boolean markAsRead)
+	public void pullEntities(final boolean markAsRead, Context context)
 		throws DaoException, ServerCommunicationException, ServerConnectionException, NoConnectionException {
 		try {
 			final AbstractAdoDao<ADO> dao = DatabaseHelper.getAdoDao(getAdoClass());
 
 			Date maxModifiedDate = dao.getLatestChangeDate();
-			Call<List<DTO>> dtoCall = pullAllSince(maxModifiedDate != null ? maxModifiedDate.getTime() : 0);
-			if (dtoCall == null) {
-				return;
-			}
-
-			Response<List<DTO>> response;
-			try {
-				response = dtoCall.execute();
-			} catch (IOException e) {
-				throw new ServerCommunicationException(e);
-			}
+			long approximateJsonSizeInBytes = getApproximateJsonSizeInBytes();
+			final Integer batchSize = approximateJsonSizeInBytes != 0
+				? RetroProvider.getNumberOfEntitiesToBePulledInOneBatch(approximateJsonSizeInBytes, context)
+				: Integer.MAX_VALUE;
+			Integer lastBatchSize = batchSize;
+
+			lastSyncedEntityDate = maxModifiedDate;
+
+			while (lastBatchSize == batchSize) {
+				Call<List<DTO>> dtoCall = pullAllSince(
+					lastSyncedEntityDate.getTime(),
+					batchSize,
+					lastSyncedEntityDate.equals(previousLastSyncedEntityDate) ? EntityDto.NO_LAST_SYNCED_UUID : lastSyncedEntityUuid);
```

Comment:
Do we really need to pass a constant here instead of null?

Reply:
yes, as it is a path parameter it cannot be optional